### ISSUE TITLE
isWebPathPlural should be inferred from remote origin

### ISSUE THREAD

**Issue by tomchkk** [Sertion/vscode-gitblame#35]
Is it not possible to infer the value of isWebPathPlural from the remote origin, rather than have to set it explicitly? I have to work with repos on bitbucket, github and gitlab, so having one value for this config item doesn't work well for me.

**Comment by Sertion** [comment]
Hey Tom!

Thank you for the issue!

When this feature first got started on I looked into the possibility of detecting what online platform someone is using. I ran into some issues:
* How do we handle slow web servers without severely impacting the user experience?
* How do we handle git repositories that does not have a web interface?
* How do we store the discovered result in a good way?
  * Using workspace settings will create a file in the repository
* Is it possible to configure the git checkout path with port and a required username and how do we convert that to the web interface url without knowing what is required and what is `git@`?
* If the detection feature does not find _system X_ it is very possible that this feature balloons out to infinity with weird exceptions and workarounds.

If you want to look into building this feature you are more than welcome to. Looking forward to the pull request!

// Albin
